### ISSUE TITLE
Val loss calculation in finetune files is incorrect

### ISSUE THREAD

**Issue by MF-FOOM** [Lightning-AI/lit-gpt#407]
For computing the train loss, the logits and targets are shifted such that output n predicts token n+1:

https://github.com/Lightning-AI/lit-gpt/blob/96d66b4845ebe287b5dd57b45e584b38d4f660e7/finetune/full.py#L165-L166

But when computing val loss, it appears this shift is forgotten, and logits and targets are just passed in raw:

https://github.com/Lightning-AI/lit-gpt/blob/96d66b4845ebe287b5dd57b45e584b38d4f660e7/finetune/full.py#L214

This results in val loss always being much much higher than it should be.

This is visible when finetuning (see how val loss is 10+ while train loss is <2? this is not overfitting, only a couple steps in):

<img width="774" alt="Screenshot 2023-08-14 at 3 08 06 PM" src="https://github.com/Lightning-AI/lit-gpt/assets/141304309/2ade5ce8-058a-40a8-8a9a-aef9450613ff">

**Comment by carmocca** [comment]
Thanks for reporting this. You are correct, I must've accidentally copypasted the wrong version in https://github.com/Lightning-AI/lit-gpt/pull/178. Would you like to open a PR with the fix?
